### DISCUSSION TITLE
Inpaint/Refine at higher resolution?

### DISCUSSION THREAD

**Discussion by rktvr** [Acly/krita-ai-diffusion#1072] [General]
is it possible to refine a selection at a higher resolution? what i mean by this is sometimes there will be something in the background i want to refine to a higher "resolution"/visual fidelity. in forge etc. i would usually just select/mask the region and then set the resolution to something like 1024x1024.

but in krita it seems like the resolution of the refine/generate is always the exact resolution of the selection?

is there a reason it is limited like this or am i misunderstanding how it works?

**Comment by minsky91** [comment]
Basically, what you've just described is what the plugin does behind the scenes. As far as I know, it will always first rescale (could be upscale or downscale, depending on the selection's pixel dimensions) the region to be refined to the native resolution of the checkpoint (512x512 for 1.5 and 1024x1024 for SDXL) at which the result can be rendered with the best quality. After being refined, the region will be rescaled back and cropped out to be fitted within the selection area. This approach allows for the  sharpest results possible, even with SDXL, which is traditionally considered a 'softer' model compared to 1.5. The resulting sharpness depends also on a number of other factors, such as sharpness of the input, the context region size, Loras and ControlNets involved, and of course the Refine Strength value, so achieving its desired degree is not a simple matter at all, but the resolution at which the region is rendered is one of the major influencers.

This is how I understand the plugin's inner workings in this regard, anyway. I have been inpainting with Krita's AI plugin for a half a year already, with image resolution varying between 1K and 16K, and it always does a fantastic job for me. Takes a bit of time to figure this all out though.

**Comment by Acly** [comment]
In addition: If you want higher resolution than native checkpoint resolution, you have to increase resolution of your canvas.

Think of it as working the other way round compared to Forget etc. You don't generate high resolution and downscale it to fit your canvas. You use a big canvas, and in the opposite case where you want faster generation it will upscale. Performance settings has various options for that (Maximum resolution, Resolution multiplier).

For reasoning see #223 #294

**Comment by rktvr** [comment]
that explains it nicely, thank you
